presto-prometheus/src/test/java/com/facebook/presto/plugin/prometheus/TestPrometheusMetricsIntegration.java

```diff
@@ -0,0 +1,121 @@
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.fail;
+
+/**
+ * Integration tests against Prometheus container
+ */
+
+@Test(priority = 1, singleThreaded = true)
+public class TestPrometheusMetricsIntegration
+        extends AbstractTestQueryFramework
+{
+    private PrometheusServer server;
+    private Session session;
+    private QueryRunner runner;
+
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        this.server = new PrometheusServer();
+        return createPrometheusQueryRunner(server);
+    }
+
+    @AfterClass(alwaysRun = true)
+    public final void destroy()
+    {
+        if (server != null) {
+            server.close();
+        }
+    }
+
+    @Test
+    public void testConfirmMetricAvailableAndCheckUp()
+            throws Exception
+    {
+        int maxTries = 60;
+        int timeBetweenTriesMillis = 1000;
+        runner = createQueryRunner();
+        session = runner.getDefaultSession();
+        int tries = 0;
+        final OkHttpClient httpClient = new OkHttpClient.Builder()
+                .connectTimeout(120, TimeUnit.SECONDS)
+                .readTimeout(120, TimeUnit.SECONDS)
+                .build();
+        HttpUrl.Builder urlBuilder = HttpUrl.parse(server.getUri().toString()).newBuilder().encodedPath("/api/v1/query");
+        urlBuilder.addQueryParameter("query", "up[1d]");
+        String url = urlBuilder.build().toString();
+        Request request = new Request.Builder()
+                .url(url)
+                .build();
+        String responseBody;
+        // this seems to be a reliable way to ensure Prometheus has `up` metric data
+        while (tries < maxTries) {
+            responseBody = httpClient.newCall(request).execute().body().string();
+            if (responseBody.contains("values")) {
+                Logger log = Logger.get(TestPrometheusMetricsIntegration.class);
+                log.info("prometheus response: %s", responseBody);
+                break;
+            }
+            Thread.sleep(timeBetweenTriesMillis);
+            tries++;
+        }
+        if (tries == maxTries) {
+            fail("Prometheus container not available for metrics query in " + maxTries * timeBetweenTriesMillis + " milliseconds.");
+        }
+        // now we're making sure the client is ready
+        tries = 0;
+        while (tries < maxTries) {
+            if (session != null && runner.tableExists(session, "up")) {
+                break;
+            }
+            Thread.sleep(timeBetweenTriesMillis);
+            tries++;
+        }
+        if (tries == maxTries) {
+            fail("Prometheus container, or client, not available for metrics query in " + maxTries * timeBetweenTriesMillis + " milliseconds.");
+        }
+
+        MaterializedResult results = runner.execute(session, "SELECT * FROM prometheus.default.up LIMIT 1").toTestTypes();
```

Comment:
Is it possible to add more tests besides `SELECT * FROM prometheus.default.up LIMIT 1` ? e.g, without limit, add timestamp range etc.

Reply:
some testing logic are implemented in `TestPrometheusRetrieveUpValueIntegrationTests`